imagebuildah/build.go

```diff
@@ -318,6 +318,37 @@ func buildDockerfilesOnce(ctx context.Context, store storage.Store, logger *logr
 		}
 		mainNode.Children = append(mainNode.Children, additionalNode.Children...)
 	}
+
+	// Check if any modifications done to labels
+	// add them to node-layer so it becomes regular
+	// layer.
+	// Reason: Docker adds label modification as
+	// last step which can be processed as regular
+	// steps and if no modification is done to layers
+	// its easier to re-use cached layers.
+	if len(options.Labels) > 0 {
+		for _, labelSpec := range options.Labels {
+			label := strings.SplitN(labelSpec, "=", 2)
+			labelLine := ""
+			key := label[0]
+			value := ""
+			if len(label) > 1 {
+				value = label[1]
+			} else {
+				value = ""
+			}
+			labelLine = fmt.Sprintf("LABEL %s=%s\n", key, value)
+			if len(labelLine) > 1 {
+				labelLineBytes := []byte(labelLine)
+				additionalNode, err := imagebuilder.ParseDockerfile(bytes.NewReader(labelLineBytes))
```

Comment:
Woudl `strings.NewReader()` work better than casting to a byte slice and using `bytes.NewReader()` here?